Issue/570-keyboard-answer-selection

SimonRothUCF opened the pull request:
Adds keyboard support for answer matching, page selection and submitting. Also adds tab support for these.

SimonRothUCF committed:
Adds keyboard accessibilty to the player

SimonRothUCF committed:
Adds tab support and accessibility fixes

clpetersonucf reviewed (changes requested):
Overall this looks good and works well. Just a few small things:

- There should probably be an indicator attached to each tile to show what keyboard shortcut is associated with it. You can see an example in the screenshot in the associated issue: https://github.com/ucfopen/last-chance-cadet-materia-widget/issues/3
- Turning the rocket or astronaut green to indicate selection is a good start, but the current selection could be a little more obvious. Maybe a small change in the tile's color or border to indicate it's selected? I'm thinking it could be the same as the hover state when you mouse over a tile.
- For genuine accessibility purposes, tab navigation should be just as functional as the keyboard shortcuts. This would mean adding styles for when a tile is focused, as well as being able to select a focused tile using `space` or `enter`. The left, right, and finish buttons would need to be given focus styles as well, and similarly work when you hit `space` or `enter`.

SimonRothUCF committed:
Adds number/letter indicators for keyboard shortcuts

SimonRothUCF committed:
Makes object focus prettier and more obvious

SimonRothUCF committed:
Allows player to select focused choice with enter

SimonRothUCF committed:
Space also works the same as enter now

SimonRothUCF committed:
Improves selection style

SimonRothUCF commented:
- I've added the number/letter indicators and they seem to be in good working order.
- I added focus styles for all of the focusable elements. The answer choices light up the same way as when you mouse over them when tabbed to. I was going to do the same for the stuff on the bottom but it didn't seem obvious enough, so for the arrows I added a light yellow outer glow and for the finish button I made the outside ring's background light red, so you can still see your progress.
- You can select things by tabbing to them and hitting enter/space now. The code for this is just a giant switch statement and could probably be prettier but it does work.
- Selecting a choice now turns the border green as well as the rocket. For some reason this does not work with already matched choices, but I'm not sure it needs to, I'm thinking that since they're all different colors anyway telling them apart won't be too much of a problem and changing the border to a completely unrelated color may actually be worse looking, and possibly more confusing, than just leaving it as the rocket only turning green, but I could also be biased due to how annoying this style has been to get working. If you think green borders on the matched choices would be best though, I will look into it.

clpetersonucf reviewed (approved):
Tested again - looks good and is functionally consistent. Code is clean and well-commented. Well done!
